Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java

```diff
@@ -166,35 +177,40 @@ public boolean onTouch(View v, MotionEvent event)
       {
         case MotionEvent.ACTION_DOWN:
         case MotionEvent.ACTION_POINTER_DOWN:
-        case MotionEvent.ACTION_MOVE:
-          // Up, Down, Left, Right
-          boolean[] pressed = {false, false, false, false};
           // If a pointer enters the bounds of a button, press that button.
           if (dpad.getBounds()
                   .contains((int) event.getX(pointerIndex), (int) event.getY(pointerIndex)))
           {
+            dpad.setTrackId(event.getPointerId(pointerIndex));
+            pressed = true;
+          }
+        case MotionEvent.ACTION_MOVE:
+          if (dpad.getTrackId() == event.getPointerId(pointerIndex))
+          {
+            // Up, Down, Left, Right
+            boolean[] dpadPressed = {false, false, false, false};
+
             if (dpad.getBounds().top + (dpad.getHeight() / 3) > (int) event.getY(pointerIndex))
-              pressed[0] = true;
+              dpadPressed[0] = true;
             if (dpad.getBounds().bottom - (dpad.getHeight() / 3) < (int) event.getY(pointerIndex))
-              pressed[1] = true;
+              dpadPressed[1] = true;
             if (dpad.getBounds().left + (dpad.getWidth() / 3) > (int) event.getX(pointerIndex))
-              pressed[2] = true;
+              dpadPressed[2] = true;
             if (dpad.getBounds().right - (dpad.getWidth() / 3) < (int) event.getX(pointerIndex))
-              pressed[3] = true;
+              dpadPressed[3] = true;
 
             // Release the buttons first, then press
-            for (int i = 0; i < pressed.length; i++)
-              if (!pressed[i])
+            for (int i = 0; i < dpadPressed.length; i++)
```

Comment:
This (and the statements that follow) didn't have curlies before, but if we apply the same standards as for C++ we should probably add some here as well (since we're here already changing things).